libcudacxx/include/cuda/__utility/static_for.h

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___UTILITY_STATIC_FOR_H
+#define _LIBCUDACXX___UTILITY_STATIC_FOR_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__utility/forward.h>
+#include <cuda/std/__utility/integer_sequence.h>
+
+#include <cuda/std/__cccl/prologue.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+template <typename _SizeType, _SizeType _Start, _SizeType _Step, typename _Operator, _SizeType... _Indices, typename... _TArgs>
+_LIBCUDACXX_HIDE_FROM_ABI constexpr void __static_for_impl(
+  _Operator __op,
+  _CUDA_VSTD::integer_sequence<_SizeType, _Indices...>,
+  _TArgs&&... __args) noexcept(noexcept(__op(_CUDA_VSTD::integral_constant<_SizeType, 0>{}, __args...)))
+{
+  (__op(_CUDA_VSTD::integral_constant<_SizeType, (_Indices * _Step + _Start)>{}, __args...), ...);
```

Comment:
No change requested: This only allows function object. Do we have any plans to expand to using `cuda::std::invoke` here?